feats/meta.py

```diff
@@ -37,18 +37,30 @@ def __init__(self, obj):
         self.description = obj.__doc__
         self.implementations: Dict[str, Implementation] = {}
         self.annotations = defaultdict(list)
-        for key in dir(obj):
-            value = getattr(obj, key)
-            if key.startswith('_') or not callable(value):
-                continue
-            impl = Implementation(value)
-            self.implementations[impl.name] = impl
-            if hasattr(value, '_feats_annotations_'):
-                for annotation in value._feats_annotations_:
-                    self.annotations[annotation].append(impl)
+
+        if callable(obj):
```

Comment:
Having Definition handle the fn here means it will have to know about the implementation of "boolean", which I don't think is desired.

What if we instead have app.boolean() create an implementation from the given function, and construct its own Definition based off of that? The definition's name, description, ect would have to come from that function.

Reply:
Good callout. This sounds good to me